web/vim_lib.js

```diff
@@ -72,6 +75,38 @@ var LibraryVIM = {
 
     dropbox: null,
 
+    // from https://github.com/s-macke/jor1k/blob/master/js/lib/utf8.js
+    // LICENSE: https://github.com/s-macke/jor1k/blob/master/LICENSE.md
+    UnicodeToUTF8Stream: function(key) {
```

Comment:
When I use Pointer_stringify, the line containing UTF-8 characters shows up blank.
It does show up if I move the cursor over the characters one by one.
It might be something about the state of the UTF-8 decoder becoming wrong, but I haven't looked into it deeply yet. I should report a bug to emscripten once I figure out the details.

I could try to call the UTF8 function directly without going through Pointer_stringify. Also, I can try the function for serializing to UTF-8 since I didn't try that one.

Yes, we could also use C code, but is more work for me since this already looks fast and is tested.